nix/e2e-tests/src/local.py

```
@@ -35,6 +35,25 @@
+
+
+def _allow_direnv_once() -> None:
+    """Allow the repo's ``.envrc`` once so run.py's ``direnv exec`` can load the
+    flake env on a fresh checkout (e.g. CI). Best-effort and idempotent; a
+    developer's already-allowed repo just gets a harmless refresh."""
+    global _direnv_allowed
+    if _direnv_allowed:
+        return
+    _direnv_allowed = True
+    try:
+        subprocess.run(
```

Ruff (0.15.15)

[error] `subprocess` call: check for execution of untrusted input

(S603)